os/arch/arm/src/amebad/amebad_irq.c

```diff
@@ -335,27 +351,31 @@ void up_irqinitialize(void)
   /* Set the priority of the SVCall interrupt */
 
 #ifdef CONFIG_ARCH_IRQPRIO
-  /* up_prioritize_irq(AMEBAD_IRQ_PENDSV, NVIC_SYSH_PRIORITY_MIN); */
+  up_prioritize_irq(AMEBAD_IRQ_PENDSV, NVIC_SYSH_PRIORITY_MIN);
 #endif
 #ifdef CONFIG_ARMV8M_USEBASEPRI
+#ifdef CONFIG_ARCH_IRQPRIO
+  up_prioritize_irq(AMEBAD_IRQ_SVCALL,NVIC_SYSH_SVCALL_PRIORITY);
+#else
   amebad_prioritize_syscall(NVIC_SYSH_SVCALL_PRIORITY);
+#endif
 #endif
 
   /* If the MPU is enabled, then attach and enable the Memory Management
    * Fault handler.
    */
-
-#ifdef CONFIG_ARM_MPU
-  //irq_attach(AMEBAD_IRQ_MEMFAULT, up_memfault, NULL);
+#if 0
+#ifdef CONFIG_ARMV8M_MPU
+  irq_attach(AMEBAD_IRQ_MEMFAULT, up_memfault, NULL);

```

Comment:
Is this not needed? There is only `ifndef` case about MPU in the below.

Reply:
fixed in 19305da

Reply:
fixed in 3e6e8b8

Reply:
fixed in 3e6e8b8